builder.py

```diff
@@ -151,101 +142,113 @@ def BuildQueueInitialize(self, queue):
+        #
+        #   1. git checkout origin/master (in each subproject)
+        #   2. Apply --branch and --subproject as needed
 
         queue.append('')
         queue.append('echo')
-        queue.append('echo ========================= Performing make distclean')
+        queue.append('echo ========================= Performing git checkout origin/master')
         queue.append('date')
-        queue.append('chmod ug+x ./configure; ./configure')
-        queue.append('make distclean')
-        queue.append('echo')
+        queue.append('git checkout origin/master')
+        queue.append('git submodule foreach git checkout origin/master')
 
-        queue.append('')
-        queue.append('if [ ${NEEDS_TFGET} -ne 0 ]; then')
-        queue.append('  echo')
-        queue.append('  echo ========================= Performing tf get')
-        queue.append('  date')
-        queue.append('  tf get')
-        queue.append('  EXITSTATUS=$?')
-        queue.append('  [ $EXITSTATUS != 0 ] && exit $EXITSTATUS')
-        queue.append('  echo')
-        queue.append('fi')
+        if self.config.options.branch:
+            queue.append('')
+            queue.append('echo')
+            queue.append('echo ========================= Performing git checkout %s' % self.config.options.branch)
+            queue.append('date')
+            queue.append('# Applying branch \'%s\' to project' % self.config.options.branch)
+            queue.append('git checkout %s || exit $?' % self.config.options.branch)
 
-        if self.config.options.shelveset:
-            shelvesetList = self.config.options.shelveset.split(',')
+        if self.config.options.subproject:
+            subprojectList = self.config.options.subproject.split(',')
             queue.append('')
             queue.append('echo')
-            queue.append('echo ========================= Performing tf unshelve')
+            queue.append('echo ========================= Performing git checkout')
             queue.append('date')
-            for shelveset in shelvesetList:
-                queue.append('echo \'Unshelving shelveset: %s\'' % shelveset)
-                queue.append('tf unshelve "%s"' % shelveset)
-                queue.append('EXITSTATUS=$?')
-                queue.append('[ $EXITSTATUS != 0 ] && exit $EXITSTATUS')
+            for subproject in subprojectList:
+                # Subproject spec looks like: <dir>:<branch>
+                subprojectElements = subproject.split(':')
+                queue.append('echo \'Applying branch %s to subproject %s\'' % (subprojectElements[1], subprojectElements[0]))
+                queue.append('if [ ! -d "%s" ]; then' % subprojectElements[0])
+                queue.append('    echo "Directory %s not found for subproject spec %s"' % (subprojectElements[0], subproject))
+                queue.append('    exit 1')
+                queue.append('fi')
+                queue.append('cd %s || exit $?' % subprojectElements[0])
```

Comment:
I did consider that at the time. After all, I do pretty much expect bash. But in the end, since I knew exactly where I wanted to be, I thought that would be safest.